### PR TITLE
Add test-keystore command to verify keystore access

### LINKED ISSUE #38: Path to verify keystore password
Forwarded from:
- https://github.com/ethereum/staking-deposit-cli/issues/170

Prior to the implementation of exit message generation, there was no mechanism with the CLI to have any form of keystore password verification.  Adding a path to verify the keystore password would be relatively trivial and not expand security concerns. Additionally, with wagyu in mind, this would allow an instant mechanism to verify keystore decryption without additional user input which could be handy when exit message generation is added.

### PR DESCRIPTION
**Pull request by valefar-on-discord**
Fixes Closes Resolves #38 

## Changes

- Adding a `test-keystore` command to verify a provided keystore file with a provided keystore password
- Updating README to fix `Language Argument` links

#### What types of changes does your code introduce?

- [X] New feature (a non-breaking change that adds functionality)
- [X] Documentation update

#### Requires testing

- [X] Yes
- [ ] No

#### If yes, did you write tests?

- [X] Yes
- [ ] No

#### Notes on testing

Providing the keystore file as an argument is easier than waiting for prompt

#### Requires documentation update

- [ ] Yes
- [X] No

#### Requires explanation in Release Notes

- [X] Yes
- [ ] No

Adding a new command `test-keystore` that will verify access to a keystore file by attempting to decrypt it with a provided keystore password. Run using `./deposit.sh test-keystore --keystore=./path/to/keystore`

### CODE REVIEW

File: ethstaker_deposit/cli/test_keystore.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        ctx: click.Context,
+        keystore: Keystore,
+        keystore_password: str,
+        **kwargs: Any) -> None:
+    click.echo(load_text(['msg_verify_keystore_file']))
+
+    try:
+        keystore.decrypt(keystore_password)
+    except ValueError:
+        click.echo(load_text(['arg_test_keystore_keystore_password', 'mismatch']))
```
Comment (remyroy) [inline ethstaker_deposit/cli/test_keystore.py:55]: Call to this echo function should include the `err=True` argument.

**Review by remyroy — CHANGES_REQUESTED** [review]
Failing to provide the `--keystore` flag and its value will prompt for the location of the keystore file. Providing a full path to that location fails with this traceback:
```
Please enter the location of your keystore file.: /home/remyroy/validator_keys/keystore-m_12381_3600_0_0_0-1725652642.json
Traceback (most recent call last):
  File "/home/remyroy/Projects/ethstaker-deposit-cli/ethstaker_deposit/deposit.py", line 132, in <module>
    run()
  File "/home/remyroy/Projects/ethstaker-deposit-cli/ethstaker_deposit/deposit.py", line 125, in run
    cli()
  File "/home/remyroy/Projects/ethstaker-deposit-cli/.venv/lib/python3.12/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/remyroy/Projects/ethstaker-deposit-cli/.venv/lib/python3.12/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/home/remyroy/Projects/ethstaker-deposit-cli/.venv/lib/python3.12/site-packages/click/core.py", line 1686, in invoke
    sub_ctx = cmd.make_context(cmd_name, args, parent=ctx)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/remyroy/Projects/ethstaker-deposit-cli/.venv/lib/python3.12/site-packages/click/core.py", line 943, in make_context
    self.parse_args(ctx, args)
  File "/home/remyroy/Projects/ethstaker-deposit-cli/.venv/lib/python3.12/site-packages/click/core.py", line 1408, in parse_args
    value, args = param.handle_parse_result(ctx, opts, args)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/remyroy/Projects/ethstaker-deposit-cli/.venv/lib/python3.12/site-packages/click/core.py", line 2400, in handle_parse_result
    value = self.process_value(ctx, value)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/remyroy/Projects/ethstaker-deposit-cli/.venv/lib/python3.12/site-packages/click/core.py", line 2356, in process_value
    value = self.type_cast_value(ctx, value)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/remyroy/Projects/ethstaker-deposit-cli/.venv/lib/python3.12/site-packages/click/core.py", line 2344, in type_cast_value
    return convert(value)
           ^^^^^^^^^^^^^^
  File "/home/remyroy/Projects/ethstaker-deposit-cli/.venv/lib/python3.12/site-packages/click/core.py", line 2316, in convert
    return self.type(value, param=self, ctx=ctx)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/remyroy/Projects/ethstaker-deposit-cli/.venv/lib/python3.12/site-packages/click/types.py", line 83, in __call__
    return self.convert(value, param, ctx)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/remyroy/Projects/ethstaker-deposit-cli/.venv/lib/python3.12/site-packages/click/types.py", line 872, in convert
    st = os.stat(rv)
         ^^^^^^^^^^^
TypeError: stat: path should be string, bytes, os.PathLike or integer, not Keystore
```

I'm guessing this is because `ethstaker_deposit/utils/validation.py:validate_keystore_file` returns a Keystore from the captive_prompt_callback and `type=click.Path` in the @jit_option expects a Path like value.

**Comment by valefar-on-discord** [comment]
Looks like the keystore prompt was an issue everywhere 😞  Fixed it across the board

**Review by remyroy — CHANGES_REQUESTED** [review]
Everything seems fine but I just noticed that running this with the `--non_interactive` flag will prompt for `Press any key.` and it shouldn't if someone wanted to automate this from another app or tool.

I don't know if we do this with every command but it would be something to look into. This could potentially be in another issue/PR.

**Comment by valefar-on-discord** [comment]
> Everything seems fine but I just noticed that running this with the `--non_interactive` flag will prompt for `Press any key.` and it shouldn't if someone wanted to automate this from another app or tool.
> 
> I don't know if we do this with every command but it would be something to look into. This could potentially be in another issue/PR.

You are correct, that is across every command we have.

Added an issue for it: https://github.com/eth-educators/ethstaker-deposit-cli/issues/157

**Pull request merged**